### PR TITLE
New Skin.ID infolabel

### PR DESCRIPTION
**Pull request by edit4ever**
adds infolabel to return active skin name
useful for addon windows conditional elements based on skin

## Description
adds infolabel to return active skin name
useful for addon windows conditional elements based on skin
also cleans up trailing whitespaces from GUIInfoManager.cpp

## Motivation and Context
Adds the ability to handle conditional includes/visibility dependent on current skin
Useful for making addons compatible with Estuary specific includes/variables

## How Has This Been Tested?
Tested on latest LibreELEC Milhouse builds

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by scott967** [comment]
I see you have done a massive cleanup on the documentation for GuiInfoManager
Could you possibly add some missing skin infolabels/bools?  Namely:
label Skin.String(name)
bool Skin.HasSetting(name) Skin.String(name) Skin.String(string1,string2)
I couldn't figure out exactly where these should be inserted in the docs so didn't try to do it myself.

**Comment by ksooo** [comment]
Could you please split your changes in two  commits - one for the  (massive) withespace cleanup, one for the new info label. Thanks.

**Comment by ksooo** [comment]
And in case you want to add missing documentation for existing labels, do this on a third commit, please or  in another PR.

File: xbmc/GUIInfoManager.cpp [hunk 1]
```diff
@@ -5931,13 +5931,19 @@ const infomap fanart_labels[] =  {{ "color1",           FANART_COLOR1 },
 ///     @skinning_v18 **[New Infolabel]** \link Skin_Font `Skin.Font`\endlink
 ///     <p>
 ///   }
+///   \table_row3{   <b>`Skin.Name`</b>,
+///                  \anchor Skin_Name
+///                  _string_,
+///     Returns the current skin name.
```
Comment (ksooo) [inline xbmc/GUIInfoManager.cpp]: Please add  "@skinning_v19" marker.

**Review by ksooo — CHANGES_REQUESTED** [review]

**Comment by ksooo** [comment]
@DaVukovic no new features in  v18, please.

**Comment by edit4ever** [comment]
As requested:
  Split commits for new infolabel and whitespace cleanup
  Added v19 marker **but LibreELEC needs this in 18.2 if possible**

**Comment by ksooo** [comment]
> Split commits for new infolabel and whitespace cleanup

 Thanks.

> LibreELEC needs this in 18.2 if possible

 For what does LE need this? How could LE need a certain Kodi feature? Kodi runs on LE, not vica versa. ;-)

**Review by ksooo — APPROVED** [review]
Thank you for your contribution.

**Comment by edit4ever** [comment]
@ksooo Thanks!  Hopefully the discussion on the LE slack channel explained the need for the addition.

BTW - I'm happy to add documentation...but this was my first Kodi PR - so could you point me to where to add that? :-)

**Comment by ksooo** [comment]
As discussed internally, the label better should be named `skin.id`, not `skin.name`.

**Review by enen92 — APPROVED** [review]
Thanks for the trailing whitespace cleanup and for adjusting the docs while contributing new features.

**Comment by ksooo** [comment]
Please squash the two functional commits down to one and adapt the PR title.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by ksooo** [comment]
@edit4ever despite from the fact that at some point in time this new label might get useful, knowing that we currently have no use case for it I suggest not to merge this until we actually need it. Every piece of code needs to be maintained and increases regression potential. Thus, I'm not a fan of dead code in the "product".

**Comment by edit4ever** [comment]
No worries - you are correct that there isn't a use case at this point.  It's here if it is ever needed down the road or just to complete the skin infolabel section. (always seemed like this was missing to me)  :-)

**Comment by phunkyfish** [comment]
PR has merit but will labelled `PR Cleanup: Backburner` and closed for now. Would be a candidate for inclusion down the line if it's required.

If the author would like it kept open please just request so on the PR.

**Pull request closed without merging**